### PR TITLE
Docker compose

### PR DESCRIPTION
**Pull request by Daerdemandt**
Solves #1063 - adds docker-compose file.

Added JSON file to describe minimal setup (web,parser,requests,retriever), this is what's used at quickstart.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -24,26 +24,17 @@ Quickstart
 ----
 * Using Docker will let you run the code in a container with all dependencies properly installed.
 * Install Docker: `curl -sSL https://get.docker.com/ | sh`
+* [Install](https://docs.docker.com/compose/install/) Docker Compose
```
Comment (howardchung) [inline README.md]: mark this optional

File: README.md [hunk 2]
```diff
@@ -24,26 +24,17 @@ Quickstart
...
-* Create tables: `sudo docker exec -i postgres psql -U postgres yasp < sql/create_tables.sql`
-* Set up Cassandra (optional): `sudo docker exec -i cassandra cqlsh < sql/cassandra.cql`
-* Get a terminal into the running container: `sudo docker exec -it yasp bash`
-* Build inside the container: `npm run build`
-* Start the services you want to run:
-  * `pm2 start pm2.json` This starts all the services according to the manifest in the JSON file.
-  * `pm2 start svc/web.js --watch` This starts a specific service and watches it for changes.
-  * `web,parser,requests,retriever` These are the minimal services for being able to open the site in a browser and request parses by ID (which is a useful end-to-end test).
-  * `pm2 logs web` You can use this command to inspect the output of a service.
+* `docker-compose up` will launch minimal setup
```
Comment (howardchung) [inline README.md]: Put all docker compose stuff in a separate section

File: README.md [hunk 3]
```diff
@@ -24,26 +24,17 @@ Quickstart
...
+* `docker exec -it yasp_web_1 bash` will give you a shell to the running container.
+  * `web,parser,requests,retriever` These are the minimal services for being able to open the site in a browser and request parses by ID (which is a useful end-to-end test). `docker/main-pm2-basic-profile.json` describes this setup and this what gets used by Compose.
+  * Other configuration that launches all services is described in `pm2.json`.
+  * `pm2 logs web` You can use this command to inspect the output of a single service. Output of all services is available at Composeer console.
```
Comment (howardchung) [inline README.md]: Composer

File: docker/main-pm2-basic-profile.json [hunk 4]
```diff
@@ -0,0 +1,37 @@
+{
```
Comment (howardchung) [inline docker/main-pm2-basic-profile.json]: thinking that profiles should go in a separate directory in the root, maybe "profiles"

Comment (Daerdemandt) [inline docker/main-pm2-basic-profile.json]: Well, I didn't want to move pm2.json - other people may already have path to it hardcoded somewhere - and creating a folder for 1 new file with minimal profile is sort of an overkill. Are there other useful profiles to keep track of?

Comment (howardchung) [inline docker/main-pm2-basic-profile.json]: The only references are in deploy.js and the readme

**Comment by Daerdemandt** [comment]
Profiles are in a separate folder, `profiles/custom.json` is gitignored.
I don't see any more generic stuff to do. Like, nginx could be added but if someone just needs a way to easily fiddle with running local instance - which looks to me as main purpose of default config - then there's no need in nginx.

**Comment by howardchung** [comment]
signoff

@albertcui review

**Pull request merged**